resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,452 @@
+
+## Existing user problem
+<!-- <Write this in first person. You basically want to summarize what “I” as a user am trying to accomplish, why the current experience is a problem and the impact it has on me, my team, my work and or biz, etc…. i.e. “When I try to do x aspect of cloud native app development, I have the following challenges / issues….<details>. Those issues result in <negative impact those challenges / issues have on your work and or business.> -->
+
+### User Jobs to be done
+
+As a platform engineer trying to build a custom platform for application development, I have the following jobs to be done and requirements to meet:
+
+#### Main job : Build a custom platform for application development
+
+|  Related jobs | Functional aspects | Emotional aspects | 
+| ------------- | ------------------ | ----------------- | 
+| Define the set of key technologies for the enterprise applications | Set of technologies that organizations intend to use for their applications based on the nature of technology, open-source support, documentation support, existing vendor relations, expertise in the team | Easy to use, troubleshoot, sense of service ownership |
+| Specify the contract for the key services in the application platform | Author a schema for the custom resources easily with auto guidance tools | Simple tools that enables the schema authoring, uniformity and consistency |
+| Iterating and testing the application platform | Productive inner loop for development and testing the application platform | Simple and easy way to iterate on changes, test and deploy them |
+| Ship the custom platform | Registering the API's to the platform, providing documentation, onboarding and support | intuitive, seamless, collaborative |
+| Maintain the custom platform | Versioning and maintaining the API's | Ease of support, sense of ownership (decision making) | 
+| Support for compliance and security | Compatibility and security policies | Trust the platform |
+| Open-source custom resources | Repository and pipeline to validate, testing the infrastructure and publishing  | Desire to contribute, easy to use, low overhead /dependency to publish |
+
+As an IT operator and developer trying to build and maintain the application, I have the following jobs to be done and requirements to meet:
+
+#### Main job : Use the custom platform to build and maintain the application
+
+|  Related jobs | Functional aspects | Emotional aspects | 
+| ------------- | ------------------ | ----------------- | 
+| Use the custom app platform to build my application  | Learn, define and consume the APIs | Easy to use, adopt, low concept count | 
+| Manage and deploy the custom resources in the application platform| Create a IaC template for the custom resource type with compliance and other policies | Easy to author, test and deploy |  
```

Comment:
I'm still unsure about including recipe authoring as part of this main job (`Use the custom platform to build and maintain the application`).

It feels better to make to include recipe authoring as part of `Build a custom platform for application development` or as its own main job.

Recipes aren't application specific, and aren't a developer responsibility. They are closer to the platform than to any application.

Reply:
separate the IT operator from the developer job